### ISSUE TITLE
JUnit Report doesn't have testcase timing details

### ISSUE THREAD

**Issue by tattah** [CleanCut/green#208]
Thanks for developing this great package. IMHO its the best Python unit testrunner that I have come across.  I have been using it for about a year along with the other tools below for the reasons given, whilst Unit, Integration and Acceptance testing:
 
* **unittest** - Preferred over nose or pytest because of compatibility reasons.
* **green** - Fast feed-back loop and sensible coloured console output.
* **xmlrunner + unittest-xml-reporting** - When I need to provide a test report/artifact in a structured format (e.g. JUnit, NUnit).
* **behave** - When I am writing BDD-style tests.   

As an enhancement, I'd like to suggest adding an output to file option (e.g. -o|--outfile *filename*) that would write the test result output to a file in an existing structured format. For instance:
```
    green -vv --outfile <my_results.xml> ./some/python/package
```
Either JUnit, NUnit2 or both would be my suggested output format as both are well known, used by many other tools and there is no real benefit creating a custom format. 

Not sure how much effort this would be but this would make green not just great for development/test development (which it already is) but also great for test reporting.

#### Benefits
* Eliminates the need to use xmlrunner + unittest-xml-reporting (which I only use because of the report generation capability).
* Eliminates the need to scrape the console (i.e. redirect stdout/err) to save a copy of the results (tried this with only partial success). 
* Easy way of providing test reports in a popular format (both locally and via CI runs).
* Complements the existing coverage capability.

**Comment by tattah** [comment]
Apologies, I just realized that green 2.16.1 has the -j|--junit-report option but I noticed that the generated JUnit report doesn't have testcase timings details.

**Comment by CleanCut** [comment]
On first glance, this might be easy by recording timing in `result.py`'s `ProtoTestResult` using `startTest` to initiate timing and `stopTest` to measure and record it.  Then the information would need to be plumbed through to the JUnit logic and hooked in.

/cc @fchauvel who contributed JUnit support and might be interested in this.

**Comment by bkmd11** [comment]
I am interested in working on this enhancement, would that be alright? I do have a question on what the end objective is. Should the output file have time taken for each individual test, or the total time for the entire test suite?

**Comment by CleanCut** [comment]
> I am interested in working on this enhancement, would that be alright?

Yes!

> Should the output file have time taken for each individual test, or the total time for the entire test suite?

I don't know the answer to that myself. I would start by looking up some example JUnit reports.

**Comment by jeansaad** [comment]
@bkmd11, I believe that the output file should have the time taken for a `<testcase>`, `<testsuite>`, and `<testsuites>`. Have a look at https://www.ibm.com/support/knowledgecenter/SSQ2R2_9.1.1/com.ibm.rsar.analysis.codereview.cobol.doc/topics/cac_useresults_junit.html for more information.

**Comment by CleanCut** [comment]
Included in Green 3.2.0 (just released).
